src/include/migraphx/op/reshape.hpp

```diff
@@ -64,45 +65,83 @@ struct reshape
 
     std::string name() const { return "reshape"; }
 
-    shape dyn_compute_shape(shape s0) const
+    // Assumes that the shape from the `dims` attribute will be valid at run-time.
+    // Makes no checks for the validity of the `dims` attribute for the given input shape.
+    shape dyn_1arg_compute_shape(shape s0) const
     {
-        auto dyn_dims = s0.dyn_dims();
-        // track number of fixed elements in input and output
-        std::size_t num_dims_ele = 1;
-        std::size_t num_dd_ele   = 1;
-        for(std::size_t i = 0; i < dyn_dims.size(); ++i)
+        auto input_dyn_dims = s0.dyn_dims();
+        bool has_negative_dim_attr = false;
```

Comment:
This can be written as `const bool has_negative_dim_attr = contains(dims, -1);`.

Reply:
Another way since you need the index of the dimension you could search for it first:

```cpp
const auto neg_dim_num = std::distance(this->dims.begin(), std::find(this->dims.begin(), this->dims.end(), -1));
const bool has_negative_dim_attr = neg_dim_num < dims.size();
```